src/setupTests.js

```diff
@@ -8,3 +8,13 @@ jest.mock('src/configStore', () => ({
 }))
 
 expect.extend(toHaveNoViolations)
+
+beforeAll(() => {
```

Comment:
I have mixed feelings about this being setup in this global place.... Normally I prefer there being an explicit reference to things like this in the tests file/s so that it is more discoverable... 

But taking a closer look, this dependency on modal-root element looks to be within our control to fix.  There are a small number of references to this element which could be replaced with a tiny helper that gets or creates the element as needed, removing this hassle entirely.

Reply:
> I have mixed feelings about this being setup in this global place.... Normally I prefer there being an explicit reference to things like this in the tests file/s so that it is more discoverable...

I don't see a gain to requiring tests to explicitly opt-in to this. I think that this makes several components work "as expected" in tests vs the current surprise that you have to mock parts of modal, create this element yourself, etc. 

But replacing the dependencies on the "modal-root" element with something that falls back to document.body sounds good to me.

Reply:
Updated to the popup components to render into document.body instead of #modal-root in https://github.com/DataBiosphere/terra-ui/pull/3485/commits/70eff4320fa31871832fd8a6a2daf4b3028b9ea6.